### ISSUE TITLE
After liquidation it is still possible to use opposing portfolio's whole `leg.basis_pos_q` by splitting it into multiple trades

### ISSUE THREAD

**Issue by julian-osec** [aeyakovenko/percolator#131]
`TradeNoCpi` doesn't scale the available `leg.basis_pos_q` amount by A-factor. This allows for trades made after liquidation of opposite side to be performed for whole leg amount by splitting the trade into a few smaller ones, small enough so they can pass under side OI limit.

Example scenario:

1. Perform deposits for 3 portfolios: long, short and a hacker.
2. Long and short open a trade for 100k
3. Long account accrues some maintenance fee and gets liquidated by `close_q = 25k`. `asset.a_short` becomes 75%
4. Short account attempts to trade their existing position to the hacker for 100k, but fails
5. Instead, short accounts does two trades by 50k and succeeds.

The resulting hacker account now has `a_basis` 75%, the same as asset's `a_short` 75%, so it is not supposed to get reduced in any subsequent operations. This effectively leaves asset with both OIs as 75k, while one long leg is 75k vs its short leg 100k.

<details><summary>Test for v16_spec_tests.rs</summary>
<p>

```rust
#[test]
fn v16_regression_trade_preflight_split_trades_total_oi() {
    let (mut header, mut markets) = market_fixture(1, 1);
    header.config.initial_margin_bps = V16PodU64::new(500);
    header.config.maintenance_margin_bps = V16PodU64::new(250);
    header.config.max_price_move_bps_per_slot = V16PodU64::new(100);
    header.config.max_trading_fee_bps = V16PodU64::new(10);

    let mut short_header = account_fixture(1, 49);
    let mut long_header = account_fixture(1, 50);
    let mut close_counterparty_header = account_fixture(1, 51);

    let raw_leg_basis_q =  5_451 * POS_SCALE + 9_451 * POS_SCALE;
    let liquidation_close_q = raw_leg_basis_q / 4;
    let adl_effective_oi_q = raw_leg_basis_q - liquidation_close_q;

    {
        let mut market = MarketGroupV16ViewMut::new(&mut header, &mut markets);
        let mut short = PortfolioV16ViewMut::new(&mut short_header);
        let mut long = PortfolioV16ViewMut::new(&mut long_header);
        let mut close_counterparty = PortfolioV16ViewMut::new(&mut close_counterparty_header);
        market.deposit_not_atomic(&mut short, 1_000).unwrap();
        market.deposit_not_atomic(&mut long, 1_000).unwrap();
        market
            .deposit_not_atomic(&mut close_counterparty, 1_000)
            .unwrap();
        market
            .execute_trade_with_fee_loss_stale_scoped_not_atomic(
                &mut long,
                &mut short,
                TradeRequestV16 {
                    asset_index: 0,
                    size_q: signed_q(raw_leg_basis_q),
                    exec_price: 1,
                    fee_bps: 0,
                },
            )
            .unwrap();
    }

    {
        let mut market = MarketGroupV16ViewMut::new(&mut header, &mut markets);
        let mut long = PortfolioV16ViewMut::new(&mut long_header);
        market
            .sync_account_fee_to_slot_not_atomic(&mut long, 100, 10_000)
            .unwrap();
        assert_eq!(long.header.capital.get(), 0);
        market
            .liquidate_account_not_atomic(
                &mut long,
                LiquidationRequestV16 {
                    asset_index: 0,
                    close_q: liquidation_close_q,
                    fee_bps: 0,
                },
            )
            .unwrap();
    }

    let asset = markets[0].engine.asset.try_to_runtime().unwrap();
    let short_leg = short_header.legs[0].try_to_runtime().unwrap();
    assert_eq!(short_leg.basis_pos_q, -(raw_leg_basis_q as i128));
    assert_eq!(asset.oi_eff_short_q, adl_effective_oi_q);
    assert_eq!(asset.a_short, 750_000_000_000_000);

    let mut market = MarketGroupV16ViewMut::new(&mut header, &mut markets);
    let mut short = PortfolioV16ViewMut::new(&mut short_header);
    let mut close_counterparty = PortfolioV16ViewMut::new(&mut close_counterparty_header);

    let err = market
        .execute_trade_with_fee_loss_stale_scoped_not_atomic(
            &mut short,
            &mut close_counterparty,
            TradeRequestV16 {
                asset_index: 0,
                size_q: signed_q(raw_leg_basis_q / 2),
                exec_price: 1,
                fee_bps: 0,
            },
        )
        .map(|_| ());
    assert_eq!(
        err,
        Ok(()),
        "trade succeds due to short having that amount in their position"
    );

    //////////////////////// second trade ////////////////////////
    {
        let err = market
            .execute_trade_with_fee_loss_stale_scoped_not_atomic(
                &mut short,
                &mut close_counterparty,
                TradeRequestV16 {
                    asset_index: 0,
                    size_q: signed_q(raw_leg_basis_q / 2),
                    exec_price: 1,
                    fee_bps: 0,
                },
            )
            .map(|_| ());
        assert_eq!(
            err,
            Ok(()), ///
            "trade preflight rejects closing the raw 14_902_000_000 short because \
            the side only has 11_176_500_000 ADL-effective short OI remaining"
        );
    }
}

```

</p>
</details>

**Comment by aeyakovenko** [comment]
Clean-room confirmation against the production SBF pinned by `percolator-meta`:

- wrapper pin: `percolator-prog@da64be639168b496833dd4c701607a94fcb06b6c`
- engine pin: `percolator@143e68c4917ed0400a27b952f036a5677047cd84`
- LiteSVM only, with ordinary public instructions and no protocol-state injection
- external backing was deposited through `TopUpBackingBucket`; the final value was converted through `ConvertReleasedPnl` and withdrawn through the real SPL-token `Withdraw` path

I ran two fresh, otherwise identical markets:

1. Deposit 1,000,000 atoms per trader and 200,000 atoms of external backing in the short-profit source domain.
2. Open balanced 0.1-lot long/short exposure at 1,000,000.
3. Publicly `RebalanceReduce` the long by half, leaving effective OI at half while the untouched short retains its full raw basis.
4. Control: keep the old short basis until the mark move. Exploit: transfer the whole raw short into a fresh portfolio as two half-sized `TradeNoCpi` calls; the one-shot whole transfer is rejected.
5. Move the authenticated mark to 500,000, drive bounded permissionless settlement, then split-transfer the profitable leg away so the claimant is flat.
6. Convert the released source claim and withdraw real collateral.

Results:

- control: opposing loss = 25,000; converted/withdrawn = 50,000; provider backing consumed = 50,000
- split path: opposing loss = 25,000; converted/withdrawn = 100,000; provider backing consumed = 100,000

The split therefore creates and withdraws an additional 50,000 atoms from honest external backing under identical market economics. This is a **real public LoF**, not only claim or points inflation.

PR #134 targets the observed root cause: its common leg-mutation gate rejects fresh attachment under non-unit A factors, and its side-scaled K/F accrual closes the related unequal-factor conservation gap. The paired wrapper PR #250 exercises the production public interface.

**Comment by aeyakovenko** [comment]
Clean-room triage against the pinned production SBF confirms the split-transfer state transition, but it does **not** establish a separate LoF vector.

I exercised the full public path with real SPL deposits and independently controlled canonical insurance: unilateral `RebalanceReduce`, two-fill transfer of the raw A-scaled short, authenticated 49% mark move, public cranks, a second post-mark two-fill transfer into a minimally collateralized relay, live PnL conversion/withdrawal attempts, matched OI flattening, permissionless stale resolution, and bounded terminal closes.

Key result:

- the malformed winner displayed `2,613,333` positive PnL;
- only `1,469,999` atoms matching independently crystallized counterparty loss became convertible and withdrawable;
- canonical insurance remained exactly `4,000,000` atoms;
- the overstated remainder could not convert before or after flattening;
- the relay's terminal close failed `Custom(25)` / `EngineCounterUnderflow`.

So the source/waterfall accounting prevents the hypothesized insurance drain. The only retained impact is the same raw-basis-versus-effective-OI terminal lock now proven publicly in #132 and Meta PR https://github.com/aeyakovenko/percolator-meta/pull/427 (commit `8b1224aa9547f9f86620d2febd6e49e272298d94`). Moving the malformed leg relocates that lock; it does not create a distinct attacker-extractable loss or an additional persistent state beyond #132.

Closing as duplicate of #132. The #132 fix must account for A-basis contributions and split-transfer rounding, so it should cover this construction too.
